app/src/main/java/org/fossasia/susi/ai/activities/MainActivity.java

```diff
@@ -922,44 +948,47 @@ private void addNewMessage(String answer, boolean isMap, boolean isHavingLink, b
-        {
+        if(answer!=null){
+            if(answer.contains("google"))
+            {
 
-            googlesearch_query = googlesearch_query.replace(" " , "+");
+                googlesearch_query = googlesearch_query.replace(" " , "+");
 
-            String url = "https://www.google.co.in/search?q=" + googlesearch_query;
+                String url = "https://www.google.co.in/search?q=" + googlesearch_query;
 
 
-            Uri uri = Uri.parse(url);
+                Uri uri = Uri.parse(url);
 
-            // create an intent builder
-            CustomTabsIntent.Builder intentBuilder = new CustomTabsIntent.Builder();
+                // create an intent builder
+                CustomTabsIntent.Builder intentBuilder = new CustomTabsIntent.Builder();
 
-            // Begin customizing
-            // set toolbar colors
-            intentBuilder.setToolbarColor(ContextCompat.getColor(getApplicationContext(), R.color.colorPrimary));
-            intentBuilder.setSecondaryToolbarColor(ContextCompat.getColor(getApplicationContext(), R.color.colorAccent));
+                // Begin customizing
+                // set toolbar colors
+                intentBuilder.setToolbarColor(ContextCompat.getColor(getApplicationContext(), R.color.colorPrimary));
+                intentBuilder.setSecondaryToolbarColor(ContextCompat.getColor(getApplicationContext(), R.color.colorAccent));
 
-            // set start and exit animations
-            //intentBuilder.setStartAnimations(this, R.anim.slide_in_right, R.anim.slide_out_left);
-            intentBuilder.setExitAnimations(getApplicationContext(), android.R.anim.slide_in_left,
-                    android.R.anim.slide_out_right);
+                // set start and exit animations
+                //intentBuilder.setStartAnimations(this, R.anim.slide_in_right, R.anim.slide_out_left);
+                intentBuilder.setExitAnimations(getApplicationContext(), android.R.anim.slide_in_left,
+                        android.R.anim.slide_out_right);
 
-            // build custom tabs intent
-            CustomTabsIntent customTabsIntent = intentBuilder.build();
+                // build custom tabs intent
+                CustomTabsIntent customTabsIntent = intentBuilder.build();
 
-            // launch the url
-            customTabsIntent.launchUrl(MainActivity.this, uri);
+                // launch the url
+                customTabsIntent.launchUrl(MainActivity.this, uri);
 
 
-            googlesearch_query = "";
-        }
+                googlesearch_query = "";
+            }
 
-        if(answer.contains("Calling")){
-            String splits[]=answer.split(" ");
-            startActivity(new Intent(Intent.ACTION_DIAL, Uri.fromParts("tel", splits[1], null)));
+            if(answer.contains("Calling")){
+                String splits[]=answer.split(" ");
+                startActivity(new Intent(Intent.ACTION_DIAL, Uri.fromParts("tel", splits[1], null)));
+            }
         }
 
+
```

Comment:
@mayank408 I think i explained the issue here: #443 Have a look.
And for your code that you implemented for Google Web Search, I would like to add something when the answer is null. If the answer will be null then, app will be getting crashed.
If you think
if(answer!=null){ ...} can avoid  indentation issues, then do suggest some ways.